client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -177,6 +211,22 @@ tie.factory('PythonCodeRunnerService', [
       }
     };
 
+    // Converts the consolidated output string into an array where the
+    // i-th entry in the array corresponds to the i-th overall test
+    // case.
+    var _createOutputArray = function(stdOutString, separator) {
+      if (stdOutString.length === 0) {
+        return [];
+      }
+      // StdOut will sometimes have extra characters at the end. Cleaning up
```

Comment:
Can you explain why? (Where do these chars come from?)

Reply:
Hm, I'm actually not too sure. It's occurring in the run somewhere since the output returned from the run already has the extra characters. I'll look into it more but it seems to be occurring on both the server version and client version with the same "garbage" at the end.

Reply:
OK. We should have an understanding of why this happens; thanks for looking into it.